apps/vanilla/src/scripts/functions.ts

```diff
@@ -0,0 +1,65 @@
+import { assertNonNull } from '@js-camp/core/utils/assertNonNull';
+
+import { AuthService } from '../services/authService';
+
+import { StorageService } from '../services/storageService';
+
+import { Token, Url } from './constants';
+import { Helpers } from './helpers';
+
+/** Check whether the user authenticated or not.*/
+export async function checkAuthentication(): Promise<boolean> {
+  const token = StorageService.get<string>(Token.Access);
+  if (token !== null) {
+    const isVerify = await AuthService.verifyToken(token);
+    if (isVerify) {
+      return true;
+    }
+    return false;
+  }
+  return false;
```

Comment:
Let's use early return pattern.
```suggestion
 if (token === null) {
   return false;
 }
 const isValid = await AuthService.verifyToken(token);
 return isValid;
```